### PR TITLE
Socket override application in loadout apply.

### LINKED ISSUE #7372: Apply socketOverrides from loadouts to items
When applying a loadout, we should also apply any saved socketOverrides to items that have them. This is mostly for configuring subclasses.

### PR DESCRIPTION
**Pull request by ryan-rushton**
This hooks up socket override application for subclasses in loadouts. 

Definitely needs some testing before it goes out. From my understanding of how things work it should be fine.

Also the logging I have in there is pretty dodgy but it will be fine while subclasses are the only items with socket overrides.

Closes #7372

### CODE REVIEW

File: src/app/loadout-drawer/loadout-apply.ts [hunk 1]
```diff
@@ -608,6 +645,80 @@ export function clearItemsOffCharacter(
...
+            const initialPlug = defs.InventoryItem.get(
+              fragmentSocket.socketDefinition.singleInitialItemHash
+            ) as PluggableInventoryItemDefinition;
+            fragmentsToClear.push({ socketIndex: fragmentSocket.socketIndex, mod: initialPlug });
+          }
+
+          // TODO more logging around the clearing of aspects.
+          // If this fails atm I just expect the following code block that assigns overrides to also
+          // fail and notify the user of it
+          await dispatch(equipMods(item.id, fragmentsToClear));
```
Comment (ryan-rushton) [inline src/app/loadout-drawer/loadout-apply.ts:687]: Note that I am specifically clearing fragments out here so we don't break the energy barrier when changing aspects. I am not sure if this is needed but figured its worth putting in initially so we can get a working version rolled out.

**Comment by bhollis** [comment]
Let's hold on this until the mods stuff works

File: src/data/d2/powerful-rewards.json [hunk 2]
```diff
@@ -1,21 +1,5 @@
...
-  2646629159,
-  3114385605,
-  3114385606,
-  3114385607,
-  3308857600,
-  3529617614,
-  3879752304
+  66825403, 73143230, 226330008, 506041342, 646467119, 782362349, 914894006, 993006552, 1204101093,
+  2223145359, 2246571627, 2484791497, 2646629159, 3114385605, 3114385606, 3114385607, 3308857600,
+  3529617614, 3879752304
```
Comment (delphiactual) [inline src/data/d2/powerful-rewards.json:4]: no

File: src/data/d2/season-to-source.json [hunk 3]
```diff
@@ -1,34 +1,9 @@
...
-    3109687656,
-    3360831066,
-    3708671066,
-    3836367751,
-    3866509906,
-    4184407433
+    18, 34, 40, 58, 268598612, 303512563, 444756050, 945330047, 1334054322, 1449602859, 1576735337,
+    1709863189, 2005599723, 2076918099, 2150402250, 2237038328, 2250046497, 2253669532, 2411768833,
+    2423200735, 3055157023, 3072652064, 3085181971, 3109687656, 3360831066, 3708671066, 3836367751,
+    3866509906, 4184407433
```
Comment (delphiactual) [inline src/data/d2/season-to-source.json:6]: no

File: src/data/d2/powerful-rewards.json (hunk 2 above)
Comment (ryan-rushton) [reply src/data/d2/powerful-rewards.json:4]: This is that bloody prettier git hook doing this. I didn't go anywhere near this file. It drives me nuts

File: src/data/d2/season-to-source.json (hunk 3 above)
Comment (ryan-rushton) [reply src/data/d2/season-to-source.json]: reverted

**Review by bhollis — APPROVED** [review]

**Comment by ryan-rushton** [comment]
Currently blocked by https://github.com/Bungie-net/api/issues/1561

**Pull request merged**